### PR TITLE
New package: nvidia-docker-2.11.0

### PR DESCRIPTION
**Pull request by quentin-z80**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me

Resolves:  #11084
Supersedes #17505

Can be tested by running:
`docker run --gpus all nvidia/cuda:9.0-base nvidia-smi`

### CODE REVIEW

**Comment by classabbyamp** [comment]
I was able to build this myself, but I was unable to run the test container successfully. The output I got was:

```
$ docker run --gpus all nvidia/cuda:9.0-base nvidia-smi  
docker: Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: detection error: open failed: /usr/lib/libnvoptix.so.1: no such file or directory: unknown.
```

I have the `nvidia` package installed, and am able to run `nvidia-smi` outside of docker with no issues. Any ideas?

**Comment by jchook** [comment]
Works for me! 🎉

At first I got this error:

```
docker: Error response from daemon: could not select device driver "" with capabilities: [[gpu]].
ERRO[0007] error waiting for container: context canceled
```

Just needed to reboot the docker daemon after installing.

**Comment by quentin-z80** [comment]
Thanks for testing it.

Do you think it would be helpful to have a install message telling the user to restart docker or is that not really needed?

**Comment by jchook** [comment]
Looks like you could add a suggestion via an `INSTALL.msg`file.

**Comment by quentin-z80** [comment]
added a post install message

File: srcpkgs/nvidia-docker/INSTALL.msg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+The docker service must be restared
```
Comment (classabbyamp) [inline srcpkgs/nvidia-docker/INSTALL.msg:1]:
```suggestion
The docker service must be restarted:
```
tiny typo :)

Comment (quentin-z80) [reply srcpkgs/nvidia-docker/INSTALL.msg]: thanks, fixed

**Comment by ryjelsum** [comment]
I was able to build this (from quentin-z80's nvidia-docker branch) and the associated dependencies and it's working well; got a massive performance increase in the Magic: the Gathering Online docker image that I am using, just to add my own feedback before this hits the main repository.

**Comment by jchook** [comment]
Is there any reason not to merge this?

**Comment by nuxion** [comment]
Hi ! I tested this package and it worked!! 
Thanks in advance for making this package available, I hope this branch will be merged soon

File: srcpkgs/libnvidia-container/template [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+makedepends="rpcsvc-proto libcap-devel libseccomp-devel libtirpc-devel"
+short_desc="NVIDIA container runtime library"
+maintainer="Quentin Freimanis <quentinfreimanis@gmail.com>"
+license="Apache-2.0"
+homepage="https://github.com/NVIDIA/libnvidia-container"
+distfiles="https://github.com/NVIDIA/libnvidia-container/archive/v${version}.tar.gz"
+checksum=5eb7c6de853307970a53e9f65cc61be7c0ceae315c175fb0a2271c5669dbf33f
+
+do_build() {
+	make dist prefix=/usr REVISION=${version}
```
Comment (classabbyamp) [inline srcpkgs/libnvidia-container/template:16]: `prefix` is patched to `/usr` in `fix_makefile.patch`, is it necessary here too?

Comment (quentin-z80) [reply srcpkgs/libnvidia-container/template:16]: removed the prefix patch and now only setting it in the template, thanks

File: srcpkgs/libnvidia-container/template [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	vinstall dist/${pkgname}_${version}/usr/lib/${pkgname}-go.so.${version} 755 /usr/lib/ ${pkgname}-go.so.1
+
+	vbin dist/${pkgname}_${version}/usr/bin/nvidia-container-cli
+}
+
+libnvidia-container-devel_package() {
+	short_desc+=" - development files"
+	depends="${sourcepkg}>=${version}_${revision}"
+	pkg_install() {
+		vinstall dist/${sourcepkg}_${version}/usr/lib/${sourcepkg}.a 644 /usr/lib/
```
Comment (classabbyamp) [inline srcpkgs/libnvidia-container/template:35]:
```suggestion
		vinstall dist/${sourcepkg}_${version}/usr/lib/${sourcepkg}.a 644 /usr/lib/
		ln -s /usr/lib/${pkgname}.so.${version} ${PKGDESTDIR}/usr/lib/${pkgname}.so
		ln -s /usr/lib/${pkgname}-go.so.${version} ${PKGDESTDIR}/usr/lib/${pkgname}-go.so
```

**Review by classabbyamp — COMMENTED** [review]
this is how things are usually done for shlibs (`.so.MAJOR -> .so.VERSION` and `.so -> .so.VERSION`)

**Pull request merged**